core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -64,14 +65,127 @@ class JDBCSourceAsColumnarStore(override val connProperties: ConnectionPropertie
   private lazy val connectionType = ExternalStoreUtils.getConnectionType(
     connProperties.dialect)
 
+  // ideally we should have only one connection saved and should be closed after commit/rollback
+  // any write ops should call beginTx first.
+  @transient
+  private implicit var conn: Connection = null
+
   override def storeColumnBatch(tableName: String, batch: ColumnBatch,
       partitionId: Int, batchId: Option[String], maxDeltaRows: Int): Unit = {
     // noinspection RedundantDefaultArgument
-    tryExecute(tableName, doInsert(tableName, batch, batchId,
-      getPartitionID(tableName, partitionId), maxDeltaRows),
-      closeOnSuccess = true, onExecutor = true)
+    //val connectedInstance = getConnectedExternalStore(tableName,onExecutor = true)
+    //connectedInstance.
+    tryExecute(tableName, closeOnSuccess = false, onExecutor = true)(doInsert(tableName, batch, batchId,
```

Comment:
Would it not be simpler if we return a connection from beginTx and use that connection to call storeColumnBatch ?